common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/RemoteBlockPushResolver.java

```diff
@@ -789,13 +789,14 @@ private void incrementIOExceptionsAndAbortIfNecessary() {
     }
 
     /**
-     * If appShuffleMergePartitionsInfo's shuffleMergeId is
+     * If appShuffleMergePartitionsInfo is null or shuffleMergeId is
      * greater than the request shuffleMergeId then it is a stale block push.
      */
     private boolean isStale(
         AppShuffleMergePartitionsInfo appShuffleMergePartitionsInfo,
         int shuffleMergeId) {
-      return appShuffleMergePartitionsInfo.shuffleMergeId > shuffleMergeId;
+      return null == appShuffleMergePartitionsInfo ||
+             appShuffleMergePartitionsInfo.shuffleMergeId > shuffleMergeId;
```

Comment:
Nit: is this indentation ok?  It should be 4

Reply:
updated.